src/main/scala/is/hail/utils/IntervalTree.scala

```diff
@@ -25,19 +25,23 @@ case class Interval(start: Any, end: Any, includeStart: Boolean, includeEnd: Boo
     compareStart > 0 && compareEnd < 0
   }
 
-  def overlaps(pord: ExtendedOrdering, other: Interval): Boolean = {
-    !this.isEmpty(pord) && !other.isEmpty(pord) &&
-      ((this.contains(pord, other.start) && (other.includeStart || this.containsWithin(pord, other.start))) ||
-      (this.contains(pord, other.end) && (other.includeEnd || this.containsWithin(pord, other.end))) ||
-      (other.contains(pord, this.start) && (this.includeStart || other.containsWithin(pord, this.start))) ||
-      (other.contains(pord, this.end) && (this.includeEnd || other.containsWithin(pord, this.end))))
+  def probablyOverlaps(pord: ExtendedOrdering, other: Interval): Boolean = {
+    !this.definitelyEmpty(pord) && !other.definitelyEmpty(pord) &&
+      ((this.contains(pord, other.start) && (other.includeStart || !pord.equiv(this.end, other.start))) ||
+        (other.contains(pord, this.start) && (this.includeStart || !pord.equiv(other.end, this.start))) ||
+        (this.contains(pord, other.end) && (other.includeEnd || !pord.equiv(this.start, other.end))) ||
+        (other.contains(pord, this.end) && (this.includeEnd || !pord.equiv(other.start, this.end))) ||
+        (pord.equiv(this.start, other.start) && pord.equiv(this.end, other.end)))
```

Comment:
This could be simplified by testing for disjointness rather than overlap, and using the "number line" ordering on intervals (I < J if I is completely to the left of J), as we discussed over lunch.